File: src/com/prey/util/StringUtil.java
package com.prey.util;

public class StringUtil {

    public static String firstCharUpper(String myString) {
            if (myString != null && myString.length() > 1) {
                    char[] stringArray = myString.toCharArray();
                    stringArray[0] = Character.toUpperCase(stringArray[0]);
                    myString = new String(stringArray);
            }
            return myString;
    }

    
    public static boolean isTextBoolean(String texto){
            boolean out=false;
            if (texto!=null){
                    texto=texto.toLowerCase().trim();
                    if ("true".equals(texto)||"false".equals(texto)){
                            return true;
                    }
            }
            return out;                
    }
    
    public static boolean isTextInteger(String texto){
            boolean out=false;
            if (texto!=null){
                    texto=texto.toLowerCase().trim();
                    try{
                            Integer.parseInt(texto);
                            return true;
                    }catch(Exception e){
                            return false;
                    }
            }
            return out;                
    }
    
    public static String classFormat(String myString){
            StringBuffer out=new StringBuffer();
            String[] array=myString.split("_");
            for(int i=0;array!=null&&i<array.length;i++){
                    out.append(firstCharUpper(array[i]));
            }
            return out.toString();
    }
    
    
    
}

File: app/src/main/java/com/prey/util/StringUtil.kt
/*******************************************************************************
 * Created by Orlando Aliaga
 * Copyright 2025 Prey Inc. All rights reserved.
 * License: GPLv3
 * Full license at "/LICENSE"
 ******************************************************************************/
package com.prey.util

import java.util.Locale

/**
 * A utility class for string manipulation.
 */
class StringUtil {

    /**
     * Converts the first character of a string to uppercase.
     *
     * @param myString the input string
     * @return the string with the first character converted to uppercase, or null if the input string is null
     */
    fun firstCharUpper(myString: String?): String? {
        var myString = myString
        if (myString != null && myString.length > 1) {
            val stringArray = myString.toCharArray()
            stringArray[0] = stringArray[0].uppercaseChar()
            myString = String(stringArray)
        }
        return myString
    }

    /**
     * Checks if a string represents a boolean value.
     *
     * @param texto the input string
     * @return true if the string represents a boolean value, false otherwise
     */
    fun isTextBoolean(texto: String?): Boolean {
        var texto = texto
        val out = false
        if (texto != null) {
            texto = texto.lowercase(Locale.getDefault()).trim { it <= ' ' }
            if ("true" == texto || "false" == texto) {
                return true
            }
        }
        return out
    }

    /**
     * Checks if a string represents an integer value.
     *
     * @param texto the input string
     * @return true if the string represents an integer value, false otherwise
     */
    fun isTextInteger(texto: String?): Boolean {
        var texto = texto
        val out = false
        if (texto != null) {
            texto = texto.lowercase(Locale.getDefault()).trim { it <= ' ' }
            try {
                texto.toInt()
                return true
            } catch (e: Exception) {
                return false
            }
        }
        return out
    }

    /**
     * Formats a string to follow the conventional Java class naming conventions.
     *
     * @param myString the input string
     * @return the formatted string
     */
    fun classFormat(myString: String): String {
        val out = StringBuffer()
        val array = myString.split("_".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        var i = 0
        while (array != null && i < array.size) {
            out.append(firstCharUpper(array[i]))
            i++
        }
        return out.toString()
    }

}